librubyfmt/src/ruby.rs

```diff
@@ -151,3 +144,21 @@ pub fn raise(s: &str) {
         rb_raise(rb_eRuntimeError, cstr.as_ptr());
     }
 }
+
+// Safety: The given VALUE must be a valid Ruby array. The lifetime is not
+// checked by the compiler. The returned slice must not outlive the given
+// Ruby array. The Ruby array must not be modified while the returned slice
+// is live.
+pub unsafe fn ruby_array_to_slice<'a>(ary: VALUE) -> &'a [VALUE] {
+    std::slice::from_raw_parts(rubyfmt_rb_ary_ptr(ary), rubyfmt_rb_ary_len(ary) as _)
```

Comment:
If we're building abstractions around this, let's make them safe(r). If you use `rubyfmt_rb_type` here to check if the type is `T_ARRAY` and return a result, that would be cleaner right?

Reply:
or if we're relying on this being called from `rb_protect`, can we document this will ruby raise and segfault if not inside an rb protect call?

Reply:
Yes, that's an additional step I'd like to take *if* we can guarantee that `VALUE` is always a valid pointer, but if we're requiring the caller to verify invariants anyway we may as well allow this one as well